projectq/meta/_compute.py

```diff
@@ -117,27 +144,9 @@ def run_uncompute(self):
         # Don't inspect each command as below -> faster uncompute
         # Just find qubits which have been allocated and deallocate them
         if len(ids_local_to_compute) == 0:
-            for cmd in reversed(self._l):
-                if cmd.gate == Allocate:
-                    qubit_id = cmd.qubits[0][0].id
-                    # Remove this qubit from MainEngine.active_qubits and
-                    # set qubit.id to = -1 in Qubit object such that it won't
-                    # send another deallocate when it goes out of scope
```

Comment:
Ah, I could tell the assignment and `__del__` method didn't both need to be there (since `__del__` also does the assignment to -1) but I kept the wrong one. Fixed in https://github.com/ProjectQ-Framework/ProjectQ/pull/109